File: implementations/node/sdk/src/contracts/executions/execution-result.ts
import type { AiSdkJsonValue } from "../../json/json-value.js";
import { AI_SDK_SCHEMA_VERSIONS } from "../common/schema-versions.js";
import type { AiSdkExecutionFailure } from "./execution-failure.js";
import type { AiSdkExecutionStatus } from "./execution-status.js";

export interface AiSdkExecutionResult {
  readonly schemaVersion: typeof AI_SDK_SCHEMA_VERSIONS.executionResult;
  readonly executionId: string;
  readonly status: AiSdkExecutionStatus;
  readonly output?: AiSdkJsonValue;
  readonly failure?: AiSdkExecutionFailure;
  readonly completedAtUtc: string;
}

File: implementations/python/sdk/src/multiplexed_ai_sdk/contracts/executions/execution_result.py
from __future__ import annotations

from dataclasses import dataclass, field

from ...json_types import AiSdkJsonValue
from ...wire_model import AiSdkWireModel
from ..common.schema_versions import AiSdkSchemaVersions
from .execution_failure import AiSdkExecutionFailure
from .execution_status import AiSdkExecutionStatus


@dataclass(frozen=True)
class AiSdkExecutionResult(AiSdkWireModel):
    execution_id: str
    status: AiSdkExecutionStatus
    completed_at_utc: str
    schema_version: int = AiSdkSchemaVersions.EXECUTION_RESULT
    output: AiSdkJsonValue | None = field(default=None, metadata={"json": True})
    failure: AiSdkExecutionFailure | None = None
